From d2db7798740061f6d54487ecaa9da15bc89cdc3c Mon Sep 17 00:00:00 2001
From: Javier Martinez Canillas <javierm@redhat.com>
Date: Tue, 28 Mar 2023 05:11:58 +0200
Subject: [PATCH] dtoverlays: Add inverted override property to
 ssd1306-spi

The new ssd130x DRM driver supports both SSD1306 I2C and SPI panels and is
compatible with the ssd1307fb driver bindings. So the "solomon,com-invdir"
DT property (to invert the COM pin scan dir) can also be used for SPI now.

This allows to configure panels whose scan direction needs to be inverted.

Signed-off-by: Javier Martinez Canillas <javierm@redhat.com>
---
 arch/arm/boot/dts/overlays/README                  | 2 ++
 arch/arm/boot/dts/overlays/ssd1306-spi-overlay.dts | 1 +
 2 files changed, 3 insertions(+)

--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -4074,6 +4074,8 @@ Params: speed                   SPI bus
         dc_pin                  GPIO pin for D/C (default 24)
         reset_pin               GPIO pin for RESET (default 25)
         height                  Display height (32 or 64; default 64)
+        inverted                Set this if display is inverted and mirrored.
+                                (default=not set)
 
 
 Name:   ssd1331-spi
--- a/arch/arm/boot/dts/overlays/ssd1306-spi-overlay.dts
+++ b/arch/arm/boot/dts/overlays/ssd1306-spi-overlay.dts
@@ -80,5 +80,6 @@
 		reset_pin = <&ssd1306>,"reset-gpios:4",
 		            <&ssd1306_pins>,"brcm,pins:0";
 		height    = <&ssd1306>,"solomon,height:0";
+		inverted = <&ssd1306>,"solomon,com-invdir?";
 	};
 };
